src/Utilities.js

```diff
@@ -76,3 +76,27 @@ function extend_(destination, source) {
   }
   return destination;
 }
+
+/* exported getValueCaseInsensitive_ */
+/**
+ * Gets the value stored in the object under the given key, in a
+ * case-insensitive way.
+ * @param {Object} obj The object to search in.
+ * @param {string} key The key to search for.
+ * @return {Object} the value under that key, or undefined otherwise
+ */
+function getValueCaseInsensitive_(obj, key) {
+  if (obj == null || typeof obj !== 'object' ||
+      key == null || !key.toString) {
+    return undefined;
+  }
+  if (key in obj) {
```

Comment:
This is a shortcut for some cases.
It's not immediately obvious why we can return here, can you add a comment?